---
layout: post
title:  Project Proposal Idea
date: '2021-01-19'
categories: hematodinium
tags: project design, in vitro
---

**Research Update - Future Plans**

I've been somewhat neglecting my notebook duties for the last while - I've really been working along two avenues, and felt like I didn't have enough in either to warrant a full post. Well, until recently, that is! This is going to be a two-part post. The first (this), will focus on how I'm progressing with planning for future research - finding a good question, determining feasibility, diving into the lit, and so on. The second will focus on how I'm progressing with my analysis of differential gene expression in Grace's _hematodinium_ samples. 

Alright. So before I begin describing my potential research project, some quick housekeeping. I imported all of Grace's thesis lit (she kindly shared her folder with me) into PaperPile, and linked each with a PDF. I'm now on the hunt for a good method for tagging and sorting all 250-odd papers!

Enough of that - here's my proposed research topic:

## Changes in gene expression for in vitro Hematodinium under different conditions

I have 5 possible variables to examine - time, temperature, region, species, and salinity. Limiting the scope to 2-3 of those variables seems much smarter. Of the 5, time/temperature seem like clear candidates. Naturally, I want to see how expression changes over time in the culture. And examining temperature means that any results are meaningful. Either a big impact of temperature or a minimal impact has interesting implications for projecting future range. 

That leaves 3 remaining variables, and I need to choose 1 (I believe). Here are the upsides and downsides of each (in brief). 

**Region**. Examining location on the local level would likely add too many confounding factors. But examining differences in expression on a regional level could help us find different ecotypes of _Hematodinium_. I have four total regions that could possibly be examined - SE Alaska, Kodiak Island/Southcentral AK, Bering Sea/Aleutian Islands, and Newfoundland. The downside is that previous studies have (as far as I know) found that Alaskan Tanner crab are panmictic, and there is minimal differentiation between the ITS1 sequences of _Hematodinium_ in Newfoundland and Alaska, so this could be chasing nothing. Another potential downside is that to keep species consistent, examining Newfoundland snow crab would require examining Bering Sea snow crab, and as described below, that can be difficult. 

**Species**. First, some background.

There are two species of Chionoecetes that are heavily commercially exploited - the Tanner crab, C. bairdi, and the snow crab, C. opilio.

The two species are closely related - in fact, they often hybridize - but have quite different patterns of distribution. There are Tanner crab populations from Southeast Alaska to the Bering Sea, but snow crab aren't present in substantial numbers outside the Bering Sea.

Despite having a wide area of overlap in the Bering Sea, infection patterns differ quite significantly in this area, as this map shows

While, I'm not sure if this is still true, here's a fascinating line from Morado et al. 2011 (emphasis mine): "although snow and Tanner crab hybrids have been encountered in the overlap area, not one hybrid has been identified or suspected of being Hematodinium positive. The number of Chionoecetes hybrids_ is not small - on the 2018 NMFS survey, they were caught at 101/375 stations, with multiple stations encountering thousands of hybrids, and the NMFS survey has had a long-term Hematodinium monitoring project incorporated (plus, late-stage Hematodinium is visually apparent). The likelihood of this being due to lack of sampling (again, if still true), is quite low.

Upsides: The seeming immunity of hybrids, along with the differences in the distribution patterns of infections, makes this quite an interesting potential subject to study.

Downsides: Snow crab primarily occur in extremely remote locations. Getting them to the lab quickly could be extremely difficult to achieve.

**Salinity**. Hematodinium appears to be affected by salinity over quite narrow ranges. In a species in the Irish Sea, infection was strongly correlated with a narrow range of salinity (34.2-34.9 ppt), and a study of Chionoecetes in Newfoundland found Hematodinium to be nearly absent below 33 ppt.
Upsides: Salinity does appear to have a substantial effect, and is quite understudied compared to other potential environmental variables.

Downsides: A study of blue crabs (which are infected by the closely-related H. perezi, not Hematodinium sp.) found that the reason for reduced infections at lower salinities was likely the inactivation of the dinospore stage (the presumptive infective stage). Cultivating dinospores in vitro is quite unpredictable, and if I'm not able to do it, the ecological significance could be limited.

The study mentioned above was looking at a wide range of salinities (5-35 ppt) in an estuarine host. As far as I know, no study has examined the effect of salinity on dinospores for a fully-marine host. Seeing if dinospores are affected by small changes in salinity would be quite interesting and impactful.

One potential way to incorporate salinity is opportunistically. If a culture happens to produce dinospores, I can divide those dinospores into various salinity treatments, expose them to slightly different salinities, and see if motility is affected. The blue crab study mentioned (Coffey et al. 2012) essentially did this, and produced some interesting results. Essentially, this would create a separate mini-experiment examining the effect of salinity on dinospores.

**Conclusions**. At this point, there's definitely more reading that I have to do, along with some investigation of feasibility. Since _Hematodinium_ is infectious, working with it in the lab could prove onerous. But at this point, I think I'm leaning towards examining time, temperature, and species, possibly with the creation of a sub-experiment investigating the effect of salinity if any cultures produce dinospores.

That's all for today - tomorrow, an update on my differential expression analysis of Grace's _Hematodinium_ transcript libraries!

